File: 4_Dicas/4.6_Code_profiling/Batalha_de_linguagens/fib.c
#include <stdio.h>

long long fib(long long n)
{
	if(n<=1)
		return n;
	return fib(n-1)+ fib(n-2);
}

int main(void)
{
	long long n = 25, result = fib(n);
	printf("Fibonacci(%lld) = %lld\n", n, result);
	return 0;
}

File: 4_Dicas/4.6_Code_profiling/Batalha_de_linguagens/fib.py
import cProfile

def fib(n):
	if n <=1:
		return n
	return fib(n-1) + fib(n-2)

def main():
	n = 25
	result = fib(n)
	print(f"Fibonacci({n}) = {result}")

cProfile.run('main()')
#main()
